lib/common/http1client.c

```diff
@@ -789,23 +817,54 @@ static void start_request(struct st_h2o_http1client_t *client, h2o_iovec_t metho
     client->_is_chunked = *props->chunked;
     client->_method_is_head = h2o_memis(method.base, method.len, H2O_STRLIT("HEAD"));
 
+    if (client->proceed_req != NULL || body.len != 0) {
```

Comment:
Do we want to indicate use of `expect: 100-continue` by adding a property `h2o_httpclient_properties_t`, rather than using a header field?

I ask this because we handle the expect header as if it was a hop-by-hop header in the server-side. The idea is that it might make sense to be consistent on both the server- and client-side.

Reply:
Basically agreed. I remember I've been struggled with this design, and there must be some reason I finally chose this way, but now I feel it's more straightforward to use h2o_httpclient_properties_t.

TBD

Reply:
Addressed in https://github.com/h2o/h2o/pull/3316/commits/ba9dd9696d9a9cca49604d69d46e37f1d4ebbd26